ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/question_answering/SentenceHighlightingQATranslator.java

```diff
@@ -107,89 +102,188 @@ public NDList processInput(TranslatorContext ctx, Input input) {
+     */
+    private Encoding getChunkEncoding(String question, String context, int chunkNumber) {
+        Encoding fullEncoding = tokenizer.encode(question, context);
+
+        if (chunkNumber == 0) {
+            return fullEncoding;
+        } else {
+            Encoding[] overflowEncodings = fullEncoding.getOverflowing();
+            if (overflowEncodings != null && chunkNumber <= overflowEncodings.length) {
+                return overflowEncodings[chunkNumber - 1];
+            } else {
+                throw new IllegalArgumentException("Invalid chunk number: " + chunkNumber);
+            }
+        }
+    }
+
+    /**
+     * Create sentence IDs array for the given chunk
+     */
+    private int[] createSentenceIdsArray(Encoding encoding, int[] wordLevelSentenceIds, int chunkNumber) {
+        long[] wordIds = encoding.getWordIds();
+        int[] sentenceIdsArray = new int[wordIds.length];
+        Arrays.fill(sentenceIdsArray, -100); // Initialize with -100 (ignore token)
+
+        // Find where the context starts in this chunk
+        long[] typeIds = encoding.getTypeIds();
+        int contextStartIndex = findContextStartIndex(typeIds);
+
+        // Map word IDs to sentence IDs
+        for (int i = contextStartIndex; i < wordIds.length; i++) {
+            long wordId = wordIds[i];
+            if (wordId != -1 && wordId < wordLevelSentenceIds.length) {
+                sentenceIdsArray[i] = wordLevelSentenceIds[(int) wordId];
+            }
+        }
+
+        return sentenceIdsArray;
+    }
+
+    /**
+     * Find where the context starts in the token sequence
+     */
+    private int findContextStartIndex(long[] typeIds) {
+        for (int i = 0; i < typeIds.length; i++) {
+            if (typeIds[i] == 1) {
+                return i;
+            }
+        }
+        return 0;  // Default to 0 if not found
+    }
+
+    /**
+     * Create model inputs from encodings and sentence IDs
+     */
+    private NDList createModelInputs(NDManager manager, Encoding encoding, int[] sentenceIdsArray) {
+        NDArray sentenceIdsNDArray = manager.create(sentenceIdsArray);
+        NDArray inputIds = manager.create(encoding.getIds());
+        NDArray attentionMask = manager.create(encoding.getAttentionMask());
+        NDArray tokenTypeIds = manager.create(encoding.getTypeIds());
+
+        sentenceIdsNDArray.setName(SENTENCE_IDS);
+        inputIds.setName(INPUT_IDS);
+        attentionMask.setName(ATTENTION_MASK);
+        tokenTypeIds.setName(TOKEN_TYPE_IDS);
+
+        return new NDList(inputIds, attentionMask, tokenTypeIds, sentenceIdsNDArray);
+    }
+
+    /**
+     * Creates an array mapping each word in the context to its sentence ID
+     */
+    private int[] createWordLevelSentenceIds(List<Sentence> sentences, String context) {
+        String[] contextWords = context.split("\\s+");
+        int[] wordSentenceIds = new int[contextWords.length];
+
+        // For each sentence
```

Comment:
should consider the case for the contextWords length 0

Reply:
the contextWords length never be 0 since we have pre validation in QuestionAnsweringInputDataSet:

```
if (context == null || context.isEmpty()) {
    throw new IllegalArgumentException("Context is not provided");
}
```